mart/attack/adversary.py

```diff
@@ -162,51 +165,44 @@ def forward(
         self,
         input: Union[torch.Tensor, tuple],
         target: Union[torch.Tensor, Dict[str, Any], tuple],
-        model: torch.nn.Module,
-        **kwargs
+        model: Optional[torch.nn.Module] = None,
+        **kwargs,
     ):
         """_summary_
 
         Args:
-            input (_type_): _description_
             target (_type_): _description_
             model (_type_): _description_
         """
 
-        self.on_run_start(self, input, target, model, **kwargs)
+        self.on_run_start(input=input, target=target, model=model, **kwargs)
 
         while True:
             try:
-                self.on_examine_start(self, input, target, model, **kwargs)
-                self.examine(input, target, model, **kwargs)
-                self.on_examine_end(self, input, target, model, **kwargs)
+                self.on_examine_start(input=input, target=target, model=model, **kwargs)
+                self.examine(input=input, target=target, model=model, **kwargs)
+                self.on_examine_end(input=input, target=target, model=model, **kwargs)
 
                 # Check the done condition here, so that every update of perturbation is examined.
                 if not self.done:
-                    self.on_advance_start(self, input, target, model, **kwargs)
-                    self.advance(input, target, model, **kwargs)
-                    self.on_advance_end(self, input, target, model, **kwargs)
+                    self.on_advance_start(input=input, target=target, model=model, **kwargs)
+                    self.advance()
+                    self.on_advance_end(input=input, target=target, model=model, **kwargs)
                     # Update cur_iter at the end so that all hooks get the correct cur_iter.
                     self.cur_iter += 1
                 else:
                     break
             except StopIteration:
                 break
 
-        self.on_run_end(self, input, target, model, **kwargs)
+        self.on_run_end(input=input, target=target, model=model, **kwargs)
 
     # Make sure we can do autograd.
     # Earlier Pytorch Lightning uses no_grad(), but later PL uses inference_mode():
     #   https://github.com/Lightning-AI/lightning/pull/12715
     @torch.enable_grad()
     @torch.inference_mode(False)
-    def examine(
-        self,
-        input: Union[torch.Tensor, tuple],
-        target: Union[torch.Tensor, Dict[str, Any], tuple],
-        model: torch.nn.Module,
-        **kwargs
-    ):
+    def examine(self, input=None, target=None, model=None, **kwargs):
```

Comment:
Can you revert this and keep the type annotations but as Optional?

Reply:
Done.